openid_connect.js

```diff
@@ -7,7 +7,38 @@ var newSession = false; // Used by oidcAuth() and validateIdToken()
 
 export default {auth, codeExchange, validateIdToken, logout};
 
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function isVarDefined(variable) {
+    const v = await sleep(10);
+    if (variable != null) return true;
+    else return false;
+}
+
+// If the ID token has not been synced yet, poll the variable every 10ms until
+// get a value or after a timeout.
+async function waitUntilVarIsDefined(r, timeout) {
```

Comment:
waitForZoneSync or waitForSessionSync? It's not as generic, as current name claims.

Reply:
Yes, changed to "waitForSessionSync"